lib/src/gauge.rs

```diff
@@ -72,3 +76,66 @@ impl StatisticSetGauge {
         })
     }
 }
+
+/// A GaugeDimensions is a group of dimensions to be used with a gauge.
+/// It is consumed when the gauge is created.
+#[derive(Debug, Default)]
+pub struct GaugeDimensions {
+    pub dimension_position: DimensionPosition,
+}
+
+impl GaugeDimensions {
+    /// Create a new, empty, GaugeDimensions
+    pub fn new() -> Self {
+        Self {
+            dimension_position: Default::default(),
+        }
+    }
+
+    /// Create a new GaugeDimensions from a given name and dimension
+    ///
+    /// ```
+    /// use goodmetrics::gauge::GaugeDimensions;
+    ///
+    /// GaugeDimensions::from("a", "dimension");
+    /// ```
+    pub fn from(name: impl Into<Name>, dimension: impl Into<Dimension>) -> Self {
+        Self {
+            dimension_position: BTreeMap::from([(name.into(), dimension.into())]),
+        }
+    }
```

Comment:
This is a copy of insert and with_dimension, semantically. It's a rename of the api in the case of the first element. I think this, as well as with_dimension should be removed.